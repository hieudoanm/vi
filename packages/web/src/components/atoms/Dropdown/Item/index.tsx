import { Menu } from '@headlessui/react';
import Link from 'next/link';
import React from 'react';

export type ItemProps = { href?: string; icon?: JSX.Element; text?: string };

const Item: React.FC<ItemProps> = ({ href = '', icon = <></>, text = '' }) => {
  return (
    <div>
      <Menu.Item>
        {({ active }) => (
          <Link href={href} passHref>
            <div className="flex items-center p-4 gap-2 cursor-pointer">
              {icon}
              <div
                className={
                  active ? 'bg-gray-100 text-gray-900' : 'text-gray-700'
                }
              >
                {text}
              </div>
            </div>
          </Link>
        )}
      </Menu.Item>
    </div>
  );
};

Item.displayName = 'Dropdown.Item';
Item.defaultProps = { href: '', icon: <></>, text: '' };

export default Item;
