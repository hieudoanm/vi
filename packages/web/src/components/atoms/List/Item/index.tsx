import React from 'react';

export type ItemProps = {
  last?: boolean;
};

const Item: React.FC<ItemProps> = ({ last, children }) => (
  <div className={last ? '' : 'border-b'}>{children}</div>
);

Item.displayName = 'List.Item';
Item.defaultProps = { last: false };

export default Item;
