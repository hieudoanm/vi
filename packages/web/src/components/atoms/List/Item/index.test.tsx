import { render } from '@testing-library/react';
import Item from '.';

describe('Item', () => {
  test('render default', () => {
    const wrapper = render(<Item>Item</Item>);
    expect(wrapper.container).toMatchSnapshot();
  });
});
