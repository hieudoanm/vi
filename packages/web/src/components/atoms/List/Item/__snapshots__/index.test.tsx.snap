// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item render default 1`] = `
<div>
  <div
    class="border-b"
  >
    Item
  </div>
</div>
`;
